opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -840,16 +833,18 @@ namespace Opm {
                 const double pvValue = ebosProblem.porosity(cell_idx) * ebosModel.dofTotalVolume( cell_idx );
                 pvSumLocal += pvValue;
 
-                for ( int phaseIdx = 0; phaseIdx < np; ++phaseIdx )
+                for (unsigned phaseIdx = 0; phaseIdx < FluidSystem::numPhases; ++phaseIdx)
                 {
-                    const int ebosPhaseIdx = flowPhaseToEbosPhaseIdx(phaseIdx);
-                    const int ebosCompIdx = flowPhaseToEbosCompIdx(phaseIdx);
+                    if (!FluidSystem::phaseIsActive(phaseIdx))
+                        continue;
+
+                    const unsigned compIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::solventComponentIndex(phaseIdx));
 
-                    B_avg[ phaseIdx ] += 1.0 / fs.invB(ebosPhaseIdx).value();
-                    const auto R2 = ebosResid[cell_idx][ebosCompIdx];
+                    B_avg[ compIdx ] += 1.0 / fs.invB(phaseIdx).value();
```

Comment:
I am confused here: B and invB both are phase properties, but here B seems to be associated with a component instead?

Reply:
The convergence is computed pr. component. and also for solvent. The vector B_avg has the same size as the residual.